drivers/can/can_stm32.c

```diff
@@ -20,10 +20,8 @@ LOG_MODULE_DECLARE(can_driver, CONFIG_CAN_LOG_LEVEL);
 
 #define CAN_INIT_TIMEOUT  (10 * sys_clock_hw_cycles_per_sec() / MSEC_PER_SEC)
 
-#if DT_HAS_NODE_STATUS_OKAY(DT_NODELABEL(can1)) && \
-	DT_NODE_HAS_COMPAT(DT_NODELABEL(can1), st_stm32_can) && \
-	DT_HAS_NODE_STATUS_OKAY(DT_NODELABEL(can2)) && \
-	DT_NODE_HAS_COMPAT(DT_NODELABEL(can2), st_stm32_can)
+#if DT_NODE_HAS_COMPAT_STATUS_OKAY(DT_NODELABEL(can1), st_stm32_can) && \
```

Comment:
While it should work, it doesn't:
`sanitycheck -b -T  samples/drivers/can/ -p olimex_stm32_p405`
```
/local/mcu/zephyrproject/zephyr/drivers/can/can_stm32.c:25:2: error: #error Simultaneous use of CAN_1 and CAN_2 not supported yet
   25 | #error Simultaneous use of CAN_1 and CAN_2 not supported yet
      |  ^~~~~

```

Reply:
@erwango fixed now, please take another look.